### PR TITLE
Adds titleDescription to TreeView

### LINKED ISSUE #105667: Add title description support to custom tree views
image

### PR DESCRIPTION
**Pull request by eamodio**
This PR fixes #105667

### CODE REVIEW

File: src/vs/vscode.proposed.d.ts [hunk 1]
```diff
@@ -2176,4 +2176,16 @@ declare module 'vscode' {
...
 	//#endregion
+
+	//#region https://github.com/microsoft/vscode/issues/105667
+
+	export interface TreeView<T> {
+		/**
+		 * An optional human-readable description that will be rendered in the title of the view.
+		 * Setting the title description to null, undefined, or empty string will remove the title description from the view.
+		 */
+		titleDescription?: string | undefined;
```
Comment (alexr00) [inline src/vs/vscode.proposed.d.ts:2187]: I think if the value is optional we don't bother with the `| undefined` in API.

**Review by alexr00 — APPROVED** [review]

File: src/vs/vscode.proposed.d.ts (hunk 1 above)
Comment (eamodio) [reply src/vs/vscode.proposed.d.ts:2187]: If I do that, then TS won't let you set `titleDescription` to `undefined` explicitly, which I kind of wanted to keep (vs just setting to an empty string) to remove the description completely.

Comment (alexr00) [reply src/vs/vscode.proposed.d.ts:2187]: I see, that makes sense.

Comment (alexr00) [inline src/vs/vscode.proposed.d.ts:2187]: Discussed at API sync. `description` is more consistent with existing API than `titleDescription`.

**Review by alexr00 — CHANGES_REQUESTED** [review]

File: src/vs/vscode.proposed.d.ts (hunk 1 above)
Comment (eamodio) [reply src/vs/vscode.proposed.d.ts]: Renamed it

**Review by alexr00 — APPROVED** [review]
Looks good!

**Pull request merged**